craco.config.js

```diff
@@ -1,19 +1,42 @@
-// eslint-disable-next-line @typescript-eslint/no-var-requires
+/* eslint-disable @typescript-eslint/no-var-requires */
 const path = require('path')
+const SentryWebpackPlugin = require('@sentry/webpack-plugin')
+const { version } = require('./package.json')
 
 // see https://github.com/gsoft-inc/craco/blob/master/packages/craco/README.md#configuration-overview
 
+const plugins = []
+const SENTRY_AUTH_TOKEN = process.env.SENTRY_AUTH_TOKEN
+const SENTRY_RELEASE_VERSION = 'CowSwap@' + version
+
+if (SENTRY_AUTH_TOKEN) {
+  plugins.push(
+    new SentryWebpackPlugin({
+      // sentry-cli configuration - can also be done directly through sentry-cli
+      // see https://docs.sentry.io/product/cli/configuration/ for details
+      authToken: SENTRY_AUTH_TOKEN,
+      org: 'gnosis-protocol',
+      project: 'cowswap',
+      release: SENTRY_RELEASE_VERSION,
+
+      // other SentryWebpackPlugin configuration
+      include: '.',
+      ignore: ['node_modules', 'webpack.config.js'],
+    })
+  )
+}
+
 module.exports = function () {
   return {
     babel: {
       plugins: ['@babel/plugin-proposal-nullish-coalescing-operator'],
     },
     webpack: {
+      plugins,
+      devtool: SENTRY_AUTH_TOKEN ? 'source-map' : '',
```

Comment:
I thought CRA would create the sourcemaps automatically unless you disable them https://stackoverflow.com/questions/55904292/how-to-generate-sourcemaps-in-create-react-app

Also, would this be disabling the sourcemap if SENTRY_AUTH_TOKEN is not present? note that if it's enabled by default, and u set '' then probably u are killing source map...

Reply:
My idea was that if we don't define auth token then we do not need/want source maps but seems like react-scripts ignore this and still create source maps. I wasn't going into too much details in the react-scripts config but seems like they are also using `source-map` value for devtool  based on this [LINK](https://github.com/facebook/create-react-app/blob/b45ae3c9caf10174d53ced1cad01a272d164f8de/packages/react-scripts/config/webpack.config.js#L183) so maybe we can remove this line, what do you suggest?

Reply:
Just as a sidenote there is a option to disable source-maps with env variable [LINK](https://github.com/facebook/create-react-app/blob/b45ae3c9caf10174d53ced1cad01a272d164f8de/packages/react-scripts/config/webpack.config.js#L42)

Reply:
My point is, if it's enabled by default, why do you need to enable it yourself with craco? 
But more concerning is, why of there's not SENTRY_AUTH_TOKEN u want to disable it? (instead of leaving the default)

In general, I would think, we always want the sourcemaps. I would leave the default if possible. If we want to disable, we can always use the official ENV var

Reply:
I removed that line. My understanding was that source maps slow build time and also load time (which seems to not be true since they only load if you open dev tools apparently).